File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/GenericShapeView.java
import java.lang.*;

	public class GenericShapeView
	{

	
		public Vector2 Position;	
		public float Radius;	
		public Vector2 Size;	
		public GenericShapeType ShapeType;	
		public static GenericShapeView Deserialize(BufferedReader input) {
		try {
			GenericShapeView _obj =  new GenericShapeView();
			// Position
			Vector2 _obj_Position = Vector2.deserialize(input);
			_obj.Position = _obj_Position;
			// Radius
			float _obj_Radius = Float.Parse(input.readLine());
			_obj.Radius = _obj_Radius;
			// Size
			Vector2 _obj_Size = Vector2.deserialize(input);
			_obj.Size = _obj_Size;
			// ShapeType
			int _obj_ShapeType = Integer.Parse(input.readLine());
			_obj.ShapeType = GenericShapeType.fromValue(_obj_ShapeType);
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
			return _obj;
		}

		public void serialize(OutputStreamWriter output) {
			try {
			// Position
			this.Position.serialize(output);
			// Radius
			output.WriteLine(((Float)this.Radius).toString() + "\n");
			// Size
			this.Size.serialize(output);
			// ShapeType
			output.append(((Integer)(this.ShapeType.getValue())).toString() + "\n");
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
		}

	}
}

File: Codinsa2015/Codinsa2015/Server/Views/GenericShapeView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views
{

	public class GenericShapeView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		/// <summary>
		/// Position de la forme : cercle => centre ; rectangle => coin supérieur gauche
		/// </summary>
		public Vector2 Position;	
		/// <summary>
		/// Si cercle : rayon du cercle.
		/// </summary>
		public float Radius;	
		/// <summary>
		/// Si rectangle : taille du rectangle.
		/// </summary>
		public Vector2 Size;	
		/// <summary>
		/// Représente le type de la forme.
		/// </summary>
		public GenericShapeType ShapeType;	
		public static GenericShapeView Deserialize(System.IO.StreamReader input) {
			GenericShapeView _obj =  new GenericShapeView();
			// Position
			Vector2 _obj_Position = Vector2.Deserialize(input);
			_obj.Position = (Vector2)_obj_Position;
			// Radius
			float _obj_Radius = Single.Parse(input.ReadLine());
			_obj.Radius = (float)_obj_Radius;
			// Size
			Vector2 _obj_Size = Vector2.Deserialize(input);
			_obj.Size = (Vector2)_obj_Size;
			// ShapeType
			GenericShapeType _obj_ShapeType = (GenericShapeType)Int32.Parse(input.ReadLine());
			_obj.ShapeType = (GenericShapeType)_obj_ShapeType;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// Position
			this.Position.Serialize(output);
			// Radius
			output.WriteLine(((float)this.Radius).ToString());
			// Size
			this.Size.Serialize(output);
			// ShapeType
			output.WriteLine(((int)this.ShapeType).ToString());
		}

	}
}
